### PR TITLE
Make sure we dont have hidden options for backend and pipeline compiler

### PR DESCRIPTION
**Pull request by 6543**
move options based on **os.Getenv** into flags

---------
*Sponsored by Kithara Software GmbH*

### CODE REVIEW

File: cli/exec/flags.go [hunk 1]
```diff
@@ -64,6 +64,25 @@ var flags = []cli.Flag{
 		Value:   "auto-detect",
 	},
 
+	//
+	// backend options for pipeline compiler
```
Comment (6543) [inline cli/exec/flags.go:68]: better name suggestions are welcome :)

other candidate of mine would be "pipeline default settings" ...

File: pipeline/backend/docker/docker.go [hunk 2]
```diff
@@ -67,20 +70,56 @@ func (e *docker) IsAvailable(context.Context) bool {
...
+	if httpClient := httpClientOfOpts(c.String("backend-docker-cert"), c.Bool("backend-docker-tls-verify")); httpClient != nil {
+		dockerClientOpts = append(dockerClientOpts, client.WithHTTPClient(httpClient))
+	}
+	if dockerHost := c.String("backend-docker-host"); dockerHost != "" {
+		dockerClientOpts = append(dockerClientOpts, client.WithHost(dockerHost))
+	}
+	if dockerAPIVersion := c.String("backend-docker-api-version"); dockerAPIVersion != "" {
+		dockerClientOpts = append(dockerClientOpts, client.WithVersion(dockerAPIVersion))
+	} else {
+		dockerClientOpts = append(dockerClientOpts, client.WithAPIVersionNegotiation())
```
Comment (6543) [inline pipeline/backend/docker/docker.go:113]: this is new and should help if we bump the agent version and still have an older docker daemon running

**Comment by anbraten** [comment]
Would be nice if you could place the flags of each backend in a flags.go file in the corresponding backend folders similar to how its done in the autoscaler: https://github.com/woodpecker-ci/autoscaler/blob/main/providers/hetznercloud/flags.go

File: shared/utils/slices.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// MergeSlices return a new slice that combines all values of input slices
+// TODO: once https://github.com/golang/go/pull/61817 got merged, we should switch to it
```
Comment (6543) [inline shared/utils/slices.go:18]: yes I submitted a pull to get this done :D
 
https://github.com/golang/go/pull/61817

**Pull request merged**